src/ray/raylet/local_object_manager.cc

```diff
@@ -298,14 +298,28 @@ void LocalObjectManager::AddSpilledUrls(
 }
 
 void LocalObjectManager::AsyncRestoreSpilledObject(
-    const ObjectID &object_id, const std::string &object_url,
+    const ObjectID &object_id, const NodeID &node_id,
     std::function<void(const ray::Status &)> callback) {
-  RAY_LOG(DEBUG) << "Restoring spilled object " << object_id << " from URL "
-                 << object_url;
   if (objects_pending_restore_.count(object_id) > 0) {
     // If the same object is restoring, we dedup here.
     return;
   }
+
+  if (!spilled_objects_url_.contains(object_id)) {
+    // If the object is not spilled locally, send a request to a remote node that spilled
+    // the object.
+    RAY_LOG(DEBUG) << "Send a object restoration request of id: " << object_id
+                   << " to a remote node: " << node_id;
+    restore_object_from_remote_node_(object_id, node_id);
```

Comment:
I think we could call this directly instead of with a callback.

Reply:
Since we anyway need some sort of callback to access remote addresses. I will keep the current structure!

Reply:
Hmm I think I would still prefer if we changed the callback to access remote addresses instead. It's just a bit confusing that this calls into the same class.

Reply:
I might be missing something, but This doesn't call into the same class right? It calls the node_manager method.